### ISSUE TITLE
Quests: Write functions that help NPC dialogue determine if the task/quest can be given

### ISSUE THREAD

**Issue by elizabethli31** [uchicago-cs/chiventure#1558]
We want to have a conditional NPC dialogue that will only be triggered if the player meets the necessary prereqs for the particular quest/task. Instead of having the NPC modules depend on prereqs, we will only need to pass ids and a qctx struct. After these functions are complete, the NPC modules can call on them to start the dialogue

**Comment by elizabethli31** [comment]
I wrote the functions for that are needed for this (npc_can_give_quest and npc_can_give_task) and just need to write tests for it now

**Comment by elizabethli31** [comment]
Wrote tests for both the functions and everything passes. In this issue, we determined functions that the NPC team will need to use in order to trigger a certain conversation. We wrote these functions, so that the NPC team will not need to use most of our structs besides the higher level ones. Now, NPC team can utilize these tests for their dialogue function. We worked with NPC who have been writing their functions using these functions already. We are continuing to check in with their progress
